projects/VitalRecord/VitalRecord_util.cs

```diff
@@ -964,124 +950,227 @@ protected void CreateAndSetPartialDate(FhirDateTime dateElement, string partUrl,
+        }
+
+        protected FhirDateTime UpdateFhirDateTimeDateElement(FhirDateTime dateElement, int? value, string partialDateUrl)
+        {
+            if (value == null)
+            {
+                return null;
+            }
+            ExtractBestDateElements(dateElement, out int? year, out int? month, out int? day, out string time);
+            // Set whichever date element we're updating to the given value.
+            switch(partialDateUrl)
+            {
+                case VR.ExtensionURL.PartialDateTimeYearVR:
+                    year = value;
+                    break;
+                case VR.ExtensionURL.PartialDateTimeMonthVR:
+                    month = value;
+                    break;
+                case VR.ExtensionURL.PartialDateTimeDayVR:
+                    day = value;
+                    break;
+                default:
+                    throw new Exception("Invalid partial date time URL");
+            }
+            return (FhirDateTime)UpdateFhirDate(year, month, day, time, true);
+        }
+
+        protected Date UpdateFhirDateTimeElement(Date dateElement, string value, bool useBirthTime = false)
+        {
+            if (value == null)
+            {
+                return null;
+            }
+            ExtractBestDateElements(dateElement, out int? year, out int? month, out int? day, out _);
+            return (Date)UpdateFhirDate(year, month, day, value, false, true);
+        }
+
+        private void ExtractBestDateElements(PrimitiveType date, out int? year, out int? month, out int? day, out string time) {
+            // Get the most valid date elements, giving priority to the parsed date elements. If the partial date is used, it will include any -1 values. If there are no valid date elements in any of the possible places, it will be null.
+            ParseDateElements(((IValue<string>)date).Value, out int? parsedYear, out int? parsedMonth, out int? parsedDay);
+            Extension pdtExt = date.GetExtension(PartialDateTimeUrl);
+            year = parsedYear ?? GetPartialDate(pdtExt, VR.ExtensionURL.PartialDateTimeYearVR);
+            month = parsedMonth ?? GetPartialDate(pdtExt, VR.ExtensionURL.PartialDateTimeMonthVR);
+            day = parsedDay ?? GetPartialDate(pdtExt, VR.ExtensionURL.PartialDateTimeDayVR);
+            time = GetTimeFragment((FhirDateTime) date.GetExtension(ExtensionURL.PatientBirthTime)?.Value)
+                    ?? ((Time)pdtExt?.GetExtension(ExtensionURL.PartialDateTimeTimeVR)?.Value)?.Value
+                    ?? GetPartialTime(date.GetExtension(PartialDateTimeUrl));
+            if (time == "unknown")
+            {
+                time = "-1";
+            }
+            else if (time == "temp-unknown")
+            {
+                time = null;
+            }
+        }
+
+        /// <summary>
+        /// Updates the Fhir Date with the given value and partial date url.
+        /// </summary>
```

Comment:
Can you add to the summary describing when this will return a Date vs a FhirDateTime?